### DISCUSSION TITLE
Very poor recognition despite good image

### DISCUSSION THREAD

**Discussion by TheNavy** [jomjol/AI-on-the-edge-device#754] [Q&A]
Does anyone have an idea how I can improve the picture from my ESP-CAM. Currently I have to reset the "preValue" every 1-2 days because it sometimes detects the wrong numbers for several hours and then the rate is too high. If I set the "MaxRateValue" higher he often detects too high values.
I already have external LEDs in use and feel the image is actually good (except for the last digit but I do not care).

Here is an Screenshot from the UI:
https://i.ibb.co/Yh1LS4X/aiontheedge.png

Thanks :)

**Comment by jomjol** [comment]
Cannot load the image. Can you update the link?

**Comment by Impact123** [comment]
The URL itself still ~~works~~worked, just copy & paste it as text into the address bar. For your convenience:
 - OP's link: [https://ibb.co/pdMLT6Q](https://ibb.co/pdMLT6Q)
 - Direct link to image: [https://i.ibb.co/Yh1LS4X/aiontheedge.png](https://i.ibb.co/Yh1LS4X/aiontheedge.png)
 
**As Attachment**
![aiontheedge](https://user-images.githubusercontent.com/899193/161619931-4c3f51c0-3007-46eb-b9a3-78b3246e1893.png)

**Reply by TheNavy** [reply]
Sorry my bad,
thanks for the correction of the link :)

Edit:
I've played with the LED Brightness a little bit but still:
![Bildschirmfoto von 2022-04-05 13-46-02](https://user-images.githubusercontent.com/34339285/161747213-04033565-ad7f-42b0-8fb6-6ab89af173ec.png)

**Reply by jomjol** [reply]
The image quality is really good (at least until the first subdigit. You can optimize your ROIs a bit: https://github.com/jomjol/AI-on-the-edge-device/wiki/ROI-Configuration

**Comment by TheNavy** [comment]
I have now changed the ROI's several times (pixel by pixel) but still have problems with it every 1-2 days.
Interestingly, the problem occurs mostly on the 5th ROI. 
Here is a screenshot from the wizard:
![Bildschirmfoto von 2022-04-12 09-22-49](https://user-images.githubusercontent.com/34339285/162904141-813a86a0-9ceb-4581-8c7f-63e4edb4a3d9.png)

**Comment by jomjol** [comment]
Then the only solution is to integrate your images into the training (see: https://github.com/jomjol/AI-on-the-edge-device/wiki/ROI-Configuration)

**Comment by wernight** [comment]
I don't have a great visual quality and it's already hard to get this quality due to the very limited space. However I wish to help find ways to improve the recognition as a human could very clearly identify the numbers.

![image](https://user-images.githubusercontent.com/145667/196518273-823242c1-d579-46c6-8bbd-fe1c539d2fd0.png)

Also having a similar issue with one where I could get a better visual quality:

![image](https://user-images.githubusercontent.com/145667/196524378-d8647188-a4f1-4d02-8936-82e7047b81bc.png)

For the second one I got better results by limiting the area for each digit to avoid the black bottom/top. However this doesn't follow the wiki suggestions.

**Reply by wernight** [reply]
Some ideas:

  - Allow to manually set a brightness/contrast (I think my hardware doesn't support it). Or even set some convolution matrix.
  - Allow to hide/block out part of the image.
  - Make it easier to manually train a network.
